src/internal.c

```diff
@@ -293,12 +293,40 @@ int wolfssl_priv_der_blind(WC_RNG* rng, DerBuffer* key, DerBuffer** mask)
     return ret;
 }
 
-void wolfssl_priv_der_unblind(DerBuffer* key, DerBuffer* mask)
+void wolfssl_priv_der_blind_toggle(DerBuffer* key, const DerBuffer* mask)
 {
     if (key != NULL) {
         xorbuf(key->buffer, mask->buffer, mask->length);
     }
 }
+
+DerBuffer *wolfssl_priv_der_unblind(const DerBuffer* key, const DerBuffer* mask)
+{
+    DerBuffer *ret;
+    if (key == NULL)
+        return NULL;
+    if (mask->length > key->length)
+        return NULL;
+    ret = (DerBuffer *)XMALLOC(sizeof(*key) + key->length, key->heap,
+                               DYNAMIC_TYPE_TMP_BUFFER);
+    if (ret == NULL)
+        return NULL;
+    XMEMCPY(ret, key, sizeof(*key));
+    ret->buffer = (byte *)ret + sizeof(*key);
```

Comment:
This kind of extra allocation at the end of `DerBuffer` is an anti-pattern. There should be an explicit field at the end of the struct to let us know that the structure stores data at the end.

Reply:
that sounds like the right idea.  I thought of checking for an existing allocator but never got around to it -- will fix.